### PR TITLE
fixes card display without a thumbnail

### PR DESCRIPTION
**Pull request by AllanOXDi**
#### Issue addressed
https://github.com/learningequality/kolibri-design-system/issues/699

## Changelog

  - **Description:**  Update`KCard` to complete vertical/horizontal layouts with no thumbnail
  - **Products impact:**  Card updates 
  - **Addresses:** https://github.com/learningequality/kolibri-design-system/pull/752
  - **Components:** `KCard` 
  - **Breaking:** N0
  - **Impacts a11y:** No
  - **Guidance:**

## Testing checklist

- [ ] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] Critical and brittle code paths are covered by unit tests
- [ ] The change is described in the changelog section above

## Reviewer guidance

- [ ] Is the code clean and well-commented?
- [ ] Are there tests for this change?
- [ ] Are all UI components LTR and RTL compliant (if applicable)?
- [ ] _Add other things to check for here_

## After review

- [ ] The changelog item has been pasted to the `CHANGELOG.md`

### CODE REVIEW

File: lib/KCard/index.vue
Comment (AlexVelezLl) [inline lib/KCard/index.vue:429]: Seems like this change is breaking the height of BelowTitle in vertical cards with large thumbnails
image

File: lib/KCard/index.vue
Comment (AlexVelezLl) [inline lib/KCard/index.vue]: Why do we need this as relative now? :eyes: Not saying that its wrong, but would like to know the reason

File: lib/KCard/index.vue
Comment (AlexVelezLl) [inline lib/KCard/index.vue]: Do we need this div if isThumbnailPresent is false? Given that in that situation, thsi div would be just simply a div without any styles I think we can safely add a v-if="isThumbnailPresent".

File: docs/pages/playground.vue
Comment (AlexVelezLl) [inline docs/pages/playground.vue]: In this example I see there is more margin in the top than in the bottom, which causes the title to be slightly misaligned.

File: docs/pages/playground.vue
Comment (AlexVelezLl) [inline docs/pages/playground.vue]: It seems that we are missing some padding at the bottom if we dont have footer.

image

**Review by AlexVelezLl — COMMENTED** [review]
Thank you @AllanOXDi! I left a couple of questions as I saw some padding issues in some examples.

**Review by MisRob — CHANGES_REQUESTED** [review]
@AllanOXDi We can't condition the presence of the thumbnail area by the presence of the thumbnail source and other factors you used to determine whether the area should be displayed. We would loose functionality that shows thumbnail placeholders when a thumbnail source is not available.

The only determining factor is `thumbnailDisplay` prop. Basically the idea is quite simple - if the thumbnail area is configured to be present via this prop, is shows no matter whether the thumbnail source is provided. If the thumbnail source is not provided in this case, the thumbnail area shows as a placeholder.

So let's just:

```
<div
  v-if="thumbnailDisplay !== ThumbnailDisplays.NONE"
  class="thumbnail"
>
```

All other conditions need to be reverted.

This is important because we quite frequently display placeholders in Kolibri. One way to observe this would be on playground - try to see what happens when you don't provide `thumbnailSrc` but still want to display the thumbnail area via `thumbnailDisplay`. Also I'd recommend you try to use `#thumbnailPlaceholder` slot.

Again, visual tests will help us greatly one day :) Until then, with every change we will need to manually preview the card in all related modes.

File: lib/KCard/index.vue
Comment (MisRob) [reply lib/KCard/index.vue]: This whole condition will need to be changed https://github.com/learningequality/kolibri-design-system/pull/752#pullrequestreview-2265059042

File: lib/KCard/index.vue
Comment (MisRob) [inline lib/KCard/index.vue]: Relatedly to my other comments, let's remove this line (see the comment on line 17-20)

File: lib/KCard/index.vue
Comment (MisRob) [inline lib/KCard/index.vue:38]: Here to we'd want to remove ` && !thumbnailSrc`.  Similar reason as other changes, see the comment on lines 34-35.

Comment (MisRob) [inline lib/KCard/index.vue]: Here too we'd want to remove ` && !thumbnailSrc`.  Similar reason as other changes, see the comment on lines 34-35.

File: lib/KCard/index.vue
Comment (MisRob) [inline lib/KCard/index.vue]: Will need to be removed in favor of checking `thumbnailDisplay` configuration. https://github.com/learningequality/kolibri-design-system/pull/752#pullrequestreview-2265059042

File: lib/KCard/index.vue
Comment (MisRob) [reply lib/KCard/index.vue:429]: Good catch, however it's fine in the scope of this PR. There's some limitations around heights that I am taking care of in https://github.com/learningequality/kolibri-design-system/pull/748 and this will to be resolved.

File: lib/KCard/index.vue
Comment (MisRob) [reply lib/KCard/index.vue]: It needs to stay `absolute`, otherwise placeholder element won't be positoned as intended via other styles right below

File: docs/pages/playground.vue
Comment (MisRob) [reply docs/pages/playground.vue]: I took care of this in https://github.com/learningequality/kolibri-design-system/pull/748

File: docs/pages/playground.vue
Comment (MisRob) [reply docs/pages/playground.vue]: I took care of this in https://github.com/learningequality/kolibri-design-system/pull/748

File: CHANGELOG.md
Comment (MisRob) [inline CHANGELOG.md:1]: Before merging, would you remove CHANGELOG.md changes from this PR completely and filled in the template in the pull request description instead? The new action will then paste it to CHANGELOG.md from there.

File: lib/KCard/index.vue
Comment (MisRob) [inline lib/KCard/index.vue]: I think here we could `v-if="thumbnailDisplay !== ThumbnailDisplays.NONE"` because we have the constants available in this component.

If you

```
data() {
  return { ThumbnailDisplays }
}
```

then the constants will become available in the template. Putting constants to `data` or `computed` is quite a common pattern in our frontend.

**Review by MisRob — APPROVED** [review]
Thank you for the updates @AllanOXDi, all good!

Just two cleanup notes that would be good to push before merging and then feel free to merge.

**Review by MisRob — CHANGES_REQUESTED** [review]
I will put changes request here temporarily so we don't merge accidentally before changelog.md cleanup finished.

**Review by MisRob — APPROVED** [review]
Thank you

**Pull request merged**
